DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskSwizzler.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// Swizzles `URLSessionTask` methods.
+internal class URLSessionTaskSwizzler {
+    static var resume: Resume?
```

Comment:
Exposing this property is unsafe as it must acquire the lock, using a `@propertyWrapper` would make it safe, or it should be private.

Reply:
actually our ReadWriteLock has something fishy and results in thread sanitizer warnings.

I moved behind a computed property with lock.

Reply:
> actually our ReadWriteLock has something fishy
sure? It's quite a simple implementation. Making the property private like you did also works 👍